torch/distributions/constraints.py

```diff
@@ -102,7 +122,49 @@ def __init__(self, low, high):
             def support(self):
                 return constraints.interval(self.low, self.high)
     """
-    pass
+    def __init__(self, fn=None, *, is_discrete=False, event_dim=0):
+        self.is_discrete = is_discrete
+        self.event_dim = event_dim
+        super().__init__(fn)
+
+    def __call__(self, fn):
+        """
+        Support for syntax to customize static attributes::
+
+            @constraints.dependent_property(is_discrete=True, event_dim=1)
+            def support(self):
+                ...
+        """
+        return _DependentProperty(fn, is_discrete=self.is_discrete, event_dim=self.event_dim)
+
+
+class _IndependentConstraint(Constraint):
+    """
+    Wraps a constraint by aggregating over ``reinterpreted_batch_ndims``-many
+    dims in :meth:`check`, so that an event is valid only if all its
+    independent entries are valid.
+    """
+    def __init__(self, base_constraint, reinterpreted_batch_ndims):
+        assert isinstance(base_constraint, Constraint)
+        assert isinstance(reinterpreted_batch_ndims, int)
```

Comment:
Do we need to unwrap if the `base_dist` is itself an instance of `Independent`? I'm asking because I'm not sure if `biject_to` will work without that.

Reply:
done